### PR TITLE
Simplified Remote Datasets

### PR DESCRIPTION
**Pull request by kozlov721**
- Simplified how remote datasets are synced
- Cleaned up the code
  - removed `dataset_id` 
  - unified access to local and remote files / metadata
- `exists`, `delete` and `list_datasets` now work with remote datasets too
- fixed bugs related to incorrect syncing
- added `detete_existing` argument to `LuxonisDataset`
  - if you want to delete remote dataset, you have to also pass `delete_remote=True` 

**Possibly breaking changes**
- removed `datasets.json` cache that was used for local storage
  - the metadata are now read from the specific datasets just like for remote

### CODE REVIEW

File: luxonis_ml/data/datasets/luxonis_dataset.py [hunk 1]
```diff
@@ -29,205 +27,134 @@
...
-            self.source = LuxonisSource("default")
-            self.datasets[self.dataset_name] = {
-                "source": self.source.to_document(),
-                "ldf_version": LDF_VERSION,
-                "classes": {},
-            }
-            self._write_datasets()
+        self.dataset_name = dataset_name
+        self.team_id = team_id or self._get_credential("LUXONISML_TEAM_ID")
+        self.team_name = team_name or self._get_credential("LUXONISML_TEAM_NAME")
```
Comment (conorsim) [inline luxonis_ml/data/datasets/luxonis_dataset.py:78]: We probably don't need the `team_name` if we want to remove it

**Review by conorsim — APPROVED** [review]

File: luxonis_ml/data/datasets/luxonis_dataset.py [hunk 2]
```diff
@@ -1,199 +1,160 @@
...
         @param bucket_type: Whether to use external cloud buckets
         @type bucket_storage: BucketStorage
         @param bucket_storage: Underlying bucket storage from local, S3, or GCS
-        @raise ValueError: If C{dataset_name} is not a string or if neither
-            C{dataset_name} nor C{dataset_id} are provided.
+        @type delete_existing: bool
+        @param delete_existing: Whether to delete a dataset with the same name if it
+            exists
+        @type delete_remote: bool
+        @param delete_remote: Whether to delete the dataset from the cloud as well
```
[the remark this answers is not in the document]
Comment (kozlov721) [reply luxonis_ml/data/datasets/luxonis_dataset.py:56]: Yeah, an extra layer of security. The reason is that when deleting from remote, all the data are truly deleted. Whereas when you deleted local dataset, it only deletes the index/parquet files etc, but the actual data are preserved.

[the remark this answers is not in the document]
Comment (tersekmatija) [reply luxonis_ml/data/datasets/luxonis_dataset.py:53]: > Whereas when you delete local dataset, it only deletes the index/parquet files etc, but the actual data are preserved.

Where is the data stored? Should we change this behavior or do we think it's reasonable for local use case?

[the remark this answers is not in the document]
Comment (kozlov721) [reply luxonis_ml/data/datasets/luxonis_dataset.py:53]: We're not copying it anywhere, so it's stored at its original location. We only store the paths.

File: luxonis_ml/utils/filesystem.py [hunk 3]
```diff
@@ -402,13 +403,20 @@ def delete_dir(
...
+        recursive: bool = True,
+        typ: Literal["file", "directory", "all"] = "file",
+    ) -> Iterator[str]:
         """Walks through the individual files in a remote directory.
 
         @type remote_dir: PathType
         @param remote_dir: Relative path to remote directory
         @type recursive: bool
         @param recursive: If True, walks through the directory recursively.
+        @type list_all: bool
```
Comment (kozlov721) [inline luxonis_ml/utils/filesystem.py:418]: Note for myself; inconsistent docstring

**Pull request merged**